src/Standards/Generic/Sniffs/WhiteSpace/LanguageConstructSpacingSniff.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Ensures all language constructs contain a single space between themselves and their content.
+ *
+ * @author    Greg Sherwood <gsherwood@squiz.net>
+ * @copyright 2006-2015 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Standards\Generic\Sniffs\WhiteSpace;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+
+class LanguageConstructSpacingSniff implements Sniff
+{
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return array(
+                T_ECHO,
+                T_PRINT,
+                T_RETURN,
+                T_INCLUDE,
+                T_INCLUDE_ONCE,
+                T_REQUIRE,
+                T_REQUIRE_ONCE,
+                T_NEW,
+                T_YIELD,
+                T_YIELD_FROM,
+                T_THROW,
+                T_NAMESPACE,
+                T_USE,
+               );
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token in
+     *                                               the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        if ($tokens[($stackPtr + 1)]['code'] === T_SEMICOLON) {
+            // No content for this language construct.
+            return;
+        }
+
+        if ($tokens[($stackPtr + 1)]['code'] === T_WHITESPACE) {
+            $content       = $tokens[($stackPtr + 1)]['content'];
+            $contentLength = strlen($content);
+            if ($contentLength !== 1) {
```

Comment:
First of all there is an issue risen here. If one of these two sniffs has changes/fixes then the sync between them can be easily forgotten.

I wonder if it would be better instead of copy-pasting the code from the Squiz sniff (as I did) the new Generic sniff to extend the Squiz sniff and only override the `register` function with more tokens.

About the corrections that you are saying I would suggest you to create a PR for the Squiz sniff and have the current PR pending until these changes are merged. Once your PR is merged we can proceed here either with the way I suggest above or by copy-pasting again the code along with the fixes. The reason is that 

 - I like to keep my PR focused on the main target and the goal of this PR was adding more tokens. It is just a way of how I prefer to work.
- Also these fixes need to be made also on the Squiz sniff.

If for those two suggestion above @gsherwood has another opinion I don't mind to go by that.